### DISCUSSION TITLE
third part client connect keycloak to login, after login it redirect succeess but return 403

### DISCUSSION THREAD

**Discussion by tracy139** [keycloak/keycloak#26688] [Q&A]
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

adapter/java-cli

### Describe the bug

I run keycloak 22.0.1 on my local mac pc by docker, then  a create third part client by springboot connect keycloak to login, after login it redirect succeess but return 403, the keycloak server log print error:

2023-08-31 02:18:29,193 WARN  [org.keycloak.events] (executor-thread-273) type=CODE_TO_TOKEN_ERROR, realmId=3bc7d689-349c-4293-8522-b817d01f665a, clientId=null, userId=null, ipAddress=172.17.0.1, error=client_not_found, grant_type=authorization_code

there is my yaml:
<img width="437" alt="image" src="https://github.com/keycloak/keycloak/assets/34903320/6b6ffd58-6b36-4ce7-a64b-cc9481e2228a">

there is my controller:
<img width="446" alt="image" src="https://github.com/keycloak/keycloak/assets/34903320/1915a678-62fd-4755-bca4-c86a514b21da">

there is my web request:
<img width="1787" alt="image" src="https://github.com/keycloak/keycloak/assets/34903320/53922761-4895-4909-8987-11002300b0ae">

http://localhost:8081/visitor/local?state=57d9a6bb-d7e0-45e3-b8e5-1672f183f3b6&session_state=88960e87-c832-4f33-b3b9-5498410f0bbf&code=5dd67639-821c-45b1-9ee6-f5e90fc6d96b.88960e87-c832-4f33-b3b9-5498410f0bbf.bd1722bb-d49a-44d8-ab7b-97288ce19a06

### Version

22.0.1

### Expected behavior

this describe has talk  clearly, please help

### Actual behavior

after login sucess redirect, it success return my controller method return 您已登录local应用

### How to Reproduce?

as i talk on describe

**Comment by ricamz** [comment]
Did you find any answer to this issue?
